### ISSUE TITLE
[ML] Forecast results show non intuitive trends 

### ISSUE THREAD

**Issue by tsviel** [elastic/ml-cpp#513]
Issue: https://github.com/elastic/ml-cpp/issues/124 was not fixed:

Hi Guys,
Unfortunately we don't really see a dramatic change when looking on the overall time series we have.
It happens for about 8.5% of our different time series (Before 7.1.0 it was 10%).

Elasticsearch version (Elastic cloud deployment v7.1.0):

Description of the problem including expected versus actual behavior:
Irrational forecasts were made to for given time series data.
Given the data it's easy for the naked eye to see that the forecasts are irrational given the history in the following examples:

https://drive.google.com/open?id=1F9pf0qSJNv9Yde2x05quASNBGilT6lCV
https://drive.google.com/open?id=1-ptdLUHP5MoyTxMnlYhhb2FpE5d9e-zP
https://drive.google.com/open?id=1EhQfqSdwLprSjxRsIxW-qF38Cf4rvtNP
https://drive.google.com/open?id=1_Lqr81TX6gcTONzj25MDFWdjGzir1XzC

Steps to reproduce:

Create index
PUT example_data/
{
"mappings": {
"properties": {
"timestamp": {
"type": "date",
"format": "epoch_second"
},
"val": {
"type": "long"
},
"series_id": {
"type": "keyword"
}
}
}
}

Index the attached example_data.json
https://drive.google.com/open?id=1OeWQs4Sg5o0_LDwMCQXN78MRVX4eAfOv
Create ml job with the following ml job json configuration.
https://drive.google.com/open?id=1C1Qovkdf6-tNJWP3L2U8M-BKrkxeqkXB
Create forecast for 8 weeks.

**Comment by tveasey** [comment]
Thanks for detailed write up. I think it is probably worth starting a separate issue for this in any case since, as you've seen, the cause for different data sets can be different. We'll take a look at the data sets you attach.

**Comment by tsviel** [comment]
Is it a separate issue :)

**Comment by tveasey** [comment]
Yes, sorry, I was observing that I think it is good thing that we closed #124 and opened a new issue since the original issue reported there was fixed.

**Comment by bondib** [comment]
Any update on this?
